libs/MeshKernel/include/MeshKernel/CurvilinearGrid/CurvilinearGrid.hpp

```diff
@@ -72,8 +71,20 @@ namespace meshkernel
         /// @brief Default constructor
         CurvilinearGrid() = default;
 
+        /// @brief Delete assignment operator
+        CurvilinearGrid& operator=(const CurvilinearGrid& other) = delete;
+
+        /// @brief Delete move assignment operator
+        CurvilinearGrid& operator=(CurvilinearGrid&& other) = delete;
+
+        /// @brief Copy constructor taking only a curvilinear grid
+        explicit CurvilinearGrid(const CurvilinearGrid& grid);
+
+        /// @brief Move constructor taking only a curvilinear grid
+        explicit CurvilinearGrid(CurvilinearGrid&& grid) = delete;
```

Comment:
why explicit?

Reply:
To avoid implicit conversions, it can also be removed

Reply:
you don't need them when the copy ctor and copy assign op are deleted